pallets/crowdloan-reward/src/lib.rs

```diff
@@ -607,13 +580,26 @@ impl<T: Config> Reward for Pallet<T>
 
     let to = <T::Lookup as StaticLookup>::unlookup(who.clone());
 
+
+    T::Currency::transfer(&from, &who, direct_reward, KeepAlive)?;
+
+    // Currently I know no way to secure that both extrinsic (transfer, vested_transfer)
+    // will be successful or be reverted if one of them changes.
+    // So, as `vested_transfer` is not revertible, we first transfer the direct amount, and then
+    // the vested amount. If first fails, we simply abort. If second fails, we are transferring
+    // the direct payout back to the module.
+    //
+    // NOTE: This procedure does change the state...
     <pallet_vesting::Module<T>>::vested_transfer(
       T::Origin::from(RawOrigin::Signed(from.clone())),
       to,
       schedule
-    )?;
+    ).map_err(|err| {
+      T::Currency::transfer(&who, &from, direct_reward, AllowDeath)
+          .err()
+          .unwrap_or_else(|| err)
+    })?;
```

Comment:
I am currently not sure if there is a better way to ensure both tx succeed or both fail.